DragGAN/DragGANApp/DragGANApp/DragGAN.cs

```diff
@@ -0,0 +1,892 @@
+
+            var size = 5;
+            using (var g = Graphics.FromImage(bmp))
+            {
+                for (int i = 0; i < handles.Count; i++)
+                {
+                    var h = handles[i];
+                    var t = targets[i];
+                    using (Brush brsh = new SolidBrush(Color.Red))
+                    {
+                        g.FillEllipse(brsh, h[0] - size, h[1] - size, size * 2, size * 2);
+                    }
+                    using (Brush brsh = new SolidBrush(Color.Blue))
+                    {
+                        g.FillEllipse(brsh, t[0] - size, t[1] - size, size * 2, size * 2);
+                    }
+                }
+            }
+            return bmp;
+        }
+
+        static public Bitmap GetImage(Tensor<float> image_data, Memory<float> pixels_memory)
+        {
+            unsafe
+            {
+                int width = image_data.Dimensions[3];
+                int height = image_data.Dimensions[2];
+                var bmp = new Bitmap(width, height, PixelFormat.Format24bppRgb);
+                var bitmapData = bmp.LockBits(new Rectangle(0, 0, width, height), ImageLockMode.ReadWrite, PixelFormat.Format24bppRgb);
+
+                byte* bits = (byte*)bitmapData.Scan0;
+                float[] pixels = pixels_memory.ToArray();
+                fixed (float* p = pixels)
+                {
+                    // Avoid having those 2 multiplications in the loop
+                    //int wtimesh1 = height * width * 1;
+                    //int wtimesh2 = height * width * 2;
+                    for (int j = 0; j < height; j++)
+                    {
+                        // Avoid multiplication in the loop
+                        int jtimesw = j * width;
+                        for (int i = 0; i < width; i++)
+                        {
+                            // unflattening 1D array using 3D coordinates
+                            //      index = z *   yMax *  xMax + y *  xMax + x
+                            int pixel = j * bitmapData.Stride + i * 3;
+                            int r_index = j * height + i;
+                            int g_index = height * width + r_index; ;
+                            int b_index = height * width + g_index;
+                            // Avoid recalculation
+                            //int jtimeswplusi = jtimesw + i;
+                            bits[pixel + 2] = (byte)MathUtils.Clamp<float>((p[r_index] * 127.5f + 127.5f), 0f, 255f);
+                            bits[pixel + 1] = (byte)MathUtils.Clamp<float>((p[g_index] * 127.5f + 127.5f), 0f, 255f);
+                            bits[pixel + 0] = (byte)MathUtils.Clamp<float>((p[b_index] * 127.5f + 127.5f), 0f, 255f);
+                        }
+                    }
+
+                }
+                bmp.UnlockBits(bitmapData);
+
+                //bmp.Save("StyleGAN.png");
+                return bmp;
+            }
+
+
+            //unsafe
```

Comment:
If not used, it should be removed. Same goes for every piece of code commented out.

Reply:
removed commented old code from both the C# and python code

Reply:
done.